src/screens/ScanQrCodeScreen.tsx

```diff
@@ -15,21 +15,27 @@ interface Props {
   navigation: StackNavigationProp<MainCardStackNavigatorParams, Route.ScanQrCode>;
   route: RouteProp<MainCardStackNavigatorParams, Route.ScanQrCode>;
 }
+
 export default class ScanQrCodeScreen extends React.PureComponent<Props> {
   state = {
-    scanned: false,
+    isBarcodeRead: false,
   };
+
   goBack = () => this.props.navigation.goBack();
-  onBarCodeScanned = (scannedQr: any) => {
-    if (this.state.scanned) {
+
+  onBarCodeScanned = (event: BarCodeReadEvent) => {
```

Comment:
I find this difficult to understand why we do need a state here and this if condition. I renamed the state name and left a comment. 

I can revert it if it doesn't make sense